app/src/components/features/rules/RulePairs/Pairs/Rows/RowsMarkup/CustomScriptRow/index.js

```diff
@@ -186,6 +187,12 @@ const CustomScriptRow = ({
 
   const handleEditorUpdate = useCallback(
     (value) => {
+      const shouldTriggerUnsavedChanges = !isFirstRender.current;
+
+      if (isFirstRender.current) {
+        isFirstRender.current = false;
+      }
+
```

Comment:
Instead of creating a `ref` to check for first render, Can we pass an optional arg `triggerUnsavedChanges` which is by default `true` and in the function call inside useEffect we pass `false`. Then we don't need to add extra logic for first render in editor update function